main.cpp

```diff
@@ -101,52 +135,116 @@ std::string Piano::displayChoosenSound(double soundKnobValue)
     return organSound;
 }
 
+double Piano::audio(int notePressed, double volume, bool isFootPedalPressed)
+{
+    if (isFootPedalPressed)
+    {
+        return 440.0 * volume;
+    }
+    return notePressed;
+}
+
 
 struct Microwave
 {
-    bool doorIsOpen = true;
-    double secondsRemaining = 0.0;
-    double power = 700.0;
-    int presetDishInput = 0;
+    Microwave();
+    bool doorIsOpen;
+    double secondsRemaining, power;
+    int presetDishInput;
 
     struct Preset
     {
-        float weightInGrams = 500.0f; 
-        int seconds = 1200;  
-        int power = 900; 
-        std::string presetName = "Potato";
+        float weightInGrams; 
+        int seconds;  
+        int power; 
+
+        Preset(float presetWeightInGrams,int presetSeconds,int presetPower);
     };
 
     void cookPotato( Preset potato );
     void cookByWeight(double selectedWeight, double power);
 };
 
+Microwave::Microwave()
+{
+    doorIsOpen = true;
+    secondsRemaining = 0.0;
+    power = 700.0;
+    presetDishInput = 0;
+    std::cout << "Instance of 'Microwave' has been constructed" << std::endl;
+}
+
+Microwave::Preset::Preset(float presetWeightInGrams,int presetSeconds,int presetPower)
+{
+    weightInGrams = presetWeightInGrams; 
+    seconds = presetSeconds;  
+    power = presetPower; 
+    std::cout << "Instance of 'Preset' has been constructed" << std::endl;
+}
+
 void Microwave::cookPotato(Microwave::Preset potato)
 {
     power = potato.power;
     secondsRemaining = potato.seconds;
+    std::cout << "Potato has been cooked" << std::endl;
 }
 
 void Microwave::cookByWeight(double selectedWeight, double selectedPower)
 {
     power = selectedPower;
     secondsRemaining = selectedWeight / selectedPower;
-}
+    std::cout << "Cooking at " << power << "w power for " << secondsRemaining << "seconds" << std::endl;
 
+}
 
 struct Airport
 {
-    int numPlanes = 50;
-    int numPassengers = 1000;
-    int numCafes = 10;
-    int coffeeStock = 500;
-    int averageMinutesInQueue = 120;
+    Airport();
+    int numPlanes, numPassengers, numCafes, coffeeStock, averageMinutesInQueue;
 
-    void boardPassengers(bool planeIsReady, int numPassengers);
+    struct Aeroplane
+    {
+        int numSeats; 
+        float fuelLevel;  
+
+        Aeroplane(int setNumSeats, float setFuelLevel);
+    };
```

Comment:
- [ ] revise: needs 5 member variables, and 3 member functions, excluding the constructor